src/acl_hostch.cpp

```diff
@@ -59,6 +59,97 @@ static cl_int l_push_packet(unsigned int physical_device_id, int channel_handle,
   }
 }
 
+static cl_int l_push_sideband_packet(unsigned int physical_device_id,
+                                     int channel_handle,
+                                     const void *host_buffer, size_t write_size,
+                                     host_pipe_t &host_pipe_info) {
+  // This pipe has sideband signals, need to break into data section and
+  // sideband signal sections
+  int status = 0;
+  bool final_status = true;
+  size_t total_pushed = 0;
+  for (auto const &sideband_signal_entry :
+       host_pipe_info.side_band_signals_vector) {
+    size_t pushed_data;
+    if (sideband_signal_entry.port_identifier == AvalonData) {
+      pushed_data = acl_get_hal()->hostchannel_push_no_ack(
+          host_pipe_info.m_physical_device_id, host_pipe_info.m_channel_handle,
+          (const void *)((char *)host_buffer +
+                         sideband_signal_entry.port_offset / 8),
+          sideband_signal_entry.side_band_size / 8, &status);
```

Comment:
Might be good to check if the offset and size is aligned to 8? Rob said this is unlikely to happen, so we are probably good without it, I'll let you decide :)